packages/providers/test/provider.test.ts

```diff
@@ -939,6 +929,82 @@ describe('Provider', () => {
     expect(estimateTxSpy).toHaveBeenCalled();
   });
 
+  it('An invalid subscription request throws a FuelError and does not hold the test runner (closes all handles)', async () => {
+    const provider = await Provider.create(FUEL_NETWORK_URL);
+
+    await expectToThrowFuelError(
+      async () => {
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
You should [prefix](https://github.com/FuelLabs/fuels-ts/blob/2a7ad0984e7f1a3822f07857784f67e4760f2ccf/.eslintrc.js#L55-L61) unused-but-required variables with `_` instead of disabling the rule.

Reply:
The rule you referenced is `no-unused-vars`, whereas this is a problem with `@typescript-eslint/naming-convention`. I tried prefixing it and the linter wouldn't allow it due to that rule. I then tried writing my own rule for it to allow it, but it doesn't seem like a quick fix and focusing too much on it would be out of the scope of this issue. I created #1352 for that purpose.

Reply:
Prefixing it with an underscore causes an issue with `@typescript-eslint/naming-convention`. I tried writing my own rule for it to allow it for a quick win, but it caused linting issues across the codebase :D. It doesn't seem like a quick fix so I created #1352. I did, however, prefix it and the rule that's now being ignored is `@typescript-eslint/naming-convention`.

Reply:
Prefixing it with an underscore causes an issue with `@typescript-eslint/naming-convention`. I tried writing my own rule for it to allow it for a quick win, but it caused linting issues across the codebase :D. It doesn't seem like a quick fix so I created #1532. I did, however, prefix it and the rule that's now being ignored is `@typescript-eslint/naming-convention`.

Reply:
Good catch, I speed-read it and didn't notice it wasn't a _function argument_. In this case, please delete the issue you created and remove the `varsIgnorePattern` from the rule, leaving only `argsIgnorePattern`. We should not allow unused _const_ declarations.

Back to your code, I will repeat myself and say I feel this `for await` looks off. If that wasn't enough, the fact you're running around lint rules to make it pass is another indication of something unusual, which I usually flag as undesirable.